src/app/ReadHandler.h

```diff
@@ -202,10 +202,10 @@ class ReadHandler : public Messaging::ExchangeDelegate
      *
      *  Constructor.
      *
-     *  The callback passed in has to outlive this handler object.
+     *  The InteractionModelEngine and the callback passed in have to outlive this handler object.
      *
      */
-    ReadHandler(ManagementCallback & apCallback, Messaging::ExchangeContext * apExchangeContext, InteractionType aInteractionType,
+    ReadHandler(InteractionModelEngine * apImEngine, ManagementCallback & apCallback, Messaging::ExchangeContext * apExchangeContext, InteractionType aInteractionType,
```

Comment:
this is public thing, constructor change may cause matter device breakage, maybe we should introduce setEngine API. @bzbarsky-apple , thought?

Reply:
Are you suggesting checking for the pointer before use and use the InteractionModelEngine::GetInstance() if it is nullptr?

Reply:
Something like that, but you should have a second Constructor, where you can populate the INE point with `InteractionModelEngine::GetInstance()`. There is code outside of the SDK that is calling possibly constructing ReadHandler using this existing Constructor, by changing it, you are breaking them

Reply:
Got it, thanks for the information. I add an extra commit to keep the constructor as it is.

Reply:
Moved the getter of the InteractionModelEngine into ManagementCallback so that the constructor stays the same.